File: JobsAPI_C#/Jobs/Jobs API/HistoricJob.cs
using System;
using System.Collections.Generic;

namespace Jobs
{
    class HistoricJob
    {
        public string deviceId { get; set; }
        public int impressions { get; set; }
        public int impressions1Color { get; set; }
        public int impressions2Colors { get; set; }
        public int impressionsNColors { get; set; }
        public string impressionsType { get; set; }
        public Ink inks { get; set; }
        public string inkUnits { get; set; }
        public string jobCompleteTime { get; set; }
        public string jobName { get; set; }
        public string jobSubmitTime { get; set; }
        public long marker { get; set; }
        public Substrate substrates { get; set; }
        public string substrateUnits { get; set; }
    }

    class HistoricJobComparator : IEqualityComparer<HistoricJob>
    {
        public bool Equals(HistoricJob x, HistoricJob y)
        {
            if (Object.ReferenceEquals(x, y)) return true;
            if (Object.ReferenceEquals(x, null) || Object.ReferenceEquals(y, null))
                return false;

            return x.marker == y.marker;
        }

        public int GetHashCode(HistoricJob job)
        {
            //Check whether the object is null
            if (Object.ReferenceEquals(job, null)) return 0;

            int hashJobId = job.marker.GetHashCode();

            //Calculate the hash code for the product.
            return hashJobId;
        }
    }
}

File: JobsAPI_PHP/HistoricJob.php
<?php
/**
 * Created by IntelliJ IDEA.
 * User: Muhannad
 * Date: 10/7/2018
 * Time: 2:10 PM
 */

namespace JobsAPI;
include "class.Ink.php";
include "class.Substrate.php";

class HistoricJob
{
  public $deviceId;
  public $impressions;
  public $impressions1Color;
  public $impressions2Colors;
  public $impressionsNColors;
  public $impressionsType;
  public $inks;
  public $inkUnits;
  public $jobCompleteTime;
  public $jobName;
  public $jobSubmitTime;
  public $marker;
  public $substrates;
  public $substrateUnits;

  public function __construct()
  {
    $this->inks = new Ink();
    $this->substrates = new Substrate();

  }

  public function __get($property)
  {
    if (property_exists($this, $property)) {
      return $this->$property;
    }
  }

  public function __set($property, $value)
  {
    if (property_exists($this, $property)) {
      $this->$property = $value;
    }

    return $this;
  }
}
